File: questao6/go/carro.go
package main

import "fmt"

// Definindo a struct Motor
type Motor struct {
    tipo     string
    potencia int
}

// Método para exibir detalhes do motor
func (m Motor) detalhesMotor() string {
    return fmt.Sprintf("Motor: %s com potência de %d cavalos", m.tipo, m.potencia)
}

// Definindo a struct Carro, que tem um Motor
type Carro struct {
    marca  string
    modelo string
    motor  Motor // Composição: Carro tem um Motor
}

// Método para exibir detalhes do carro
func (c Carro) detalhesCarro() string {
    return fmt.Sprintf("Carro: %s %s, %s", c.marca, c.modelo, c.motor.detalhesMotor())
}

func main() {
    // Criando um motor
    motor1 := Motor{"V8", 500}

    // Criando um carro com o motor associado
    carro1 := Carro{"Ford", "Mustang", motor1}

    // Exibindo os detalhes do carro
    fmt.Println(carro1.detalhesCarro())
}

File: questao6/python/carro.py
# Definindo a classe Motor
class Motor:
    def __init__(self, tipo, potencia):
        self.tipo = tipo
        self.potencia = potencia

    def detalhes_motor(self):
        return f"Motor: {self.tipo} com potência de {self.potencia} cavalos"


# Definindo a classe Carro
class Carro:
    def __init__(self, marca, modelo, motor):
        self.marca = marca
        self.modelo = modelo
        self.motor = motor  # Composição: Carro tem um Motor

    def detalhes_carro(self):
        return f"Carro: {self.marca} {self.modelo}, {self.motor.detalhes_motor()}"


# Criando um objeto Motor
motor1 = Motor("V8", 500)

# Criando um objeto Carro com o motor associado
carro1 = Carro("Ford", "Mustang", motor1)

# Exibindo os detalhes do carro, incluindo os detalhes do motor
print(carro1.detalhes_carro())
